src/vma/event/event_handler_manager.cpp

```diff
@@ -152,7 +152,9 @@ void event_handler_manager::unregister_timers_event_and_delete(timer_handler* ha
 	memset(&reg_action, 0, sizeof(reg_action));
 	reg_action.type = UNREGISTER_TIMERS_AND_DELETE;
 	reg_action.info.timer.handler = handler;
-	post_new_reg_action(reg_action);
+	if (!post_new_reg_action(reg_action)) {
+		delete handler;
```

Comment:
I think it should be inherited from `cleanable_obj` that is not destroyed explicitly.